src/mission.cpp

```diff
@@ -2517,7 +2517,7 @@ DROID *buildMissionDroid(DROID_TEMPLATE *psTempl, UDWORD x, UDWORD y, UDWORD pla
 	psNewDroid->pos.y = INVALID_XY;
 
 	//set all the droids to selected from when return
-	psNewDroid->selected = true;
+	psNewDroid->selected = !isTransporter(psNewDroid);
```

Comment:
Actually, is it worth pulling out the tests in `SelectDroid(DROID *psDroid)` to a new `isSelectable` function, like below?

```
bool isSelectable(DROID const *psDroid)
{
	if (psDroid->flags.test(OBJECT_FLAG_UNSELECTABLE))
	{
		return false;
	}
	// we shouldn't ever control the transporter in SP games
	if (isTransporter(psDroid) && !bMultiPlayer)
	{
		return false;
	}
	return true;
}
```

We could then use it here, and in `SelectDroid`.

Reply:
Sure, why not.